reactive/flannel.py

```diff
@@ -1,254 +1,190 @@
+        'Backend': {
+            'Type': 'vxlan'
+        }
+    })
+    cmd = "etcdctl "
+    cmd += "--endpoint '{0}' ".format(etcd.get_connection_string())
+    cmd += "--cert-file {0} ".format(ETCD_CERT_PATH)
+    cmd += "--key-file {0} ".format(ETCD_KEY_PATH)
+    cmd += "--ca-file {0} ".format(ETCD_CA_PATH)
+    cmd += "set /coreos.com/network/config '{0}'".format(data)
+    check_call(split(cmd))
+    set_state('flannel.network.configured')
+
+
+@when('flannel.binaries.installed', 'flannel.service.installed',
+      'flannel.network.configured')
+@when_not('flannel.service.started')
+def start_flannel_service():
+    ''' Start the flannel service. '''
+    status_set('maintenance', 'Starting flannel service.')
+    service_start('flannel')
+    set_state('flannel.service.started')
+
+
+@when('flannel.service.started', 'cni.is-worker')
+@when_not('flannel.cni.available')
+def set_available(cni):
+    ''' Indicate to the CNI provider that we're ready. '''
+    cni.set_available()
+    set_state('flannel.cni.available')
+
+
+@when('flannel.binaries.installed')
+@when_not('flannel.version.set')
 def set_flannel_version():
     ''' Surface the currently deployed version of flannel to Juju '''
-    # flanneld -version
-    # v0.6.1
+    status_set('maintenance', 'Setting flannel version.')
     cmd = 'flanneld -version'
-    p = Popen(cmd, shell=True,
-              stdin=PIPE,
-              stdout=PIPE,
-              stderr=STDOUT,
-              close_fds=True)
-    version = p.stdout.read()
+    version = check_output(split(cmd), stderr=STDOUT).decode('utf-8')
     if version:
-        hookenv.application_version_set(version.split(b'v')[-1].rstrip())
+        application_version_set(version.split('v')[-1].strip())
+        set_state('flannel.version.set')
 
 
-@when('flannel.sdn.configured', 'host.connected')
-@when_not('flannel.host.relayed')
-def relay_sdn_configuration(plugin_host):
-    ''' send the flannel interface configuration to the principal unit '''
-    try:
-        subnet, mtu = _ingest_network_config()
-        cidr = hookenv.config('cidr')
-        plugin_host.set_configuration(mtu, subnet, cidr)
-        set_state('flannel.host.relayed')
-        hookenv.status_set('active', 'Flannel subnet {0}'.format(subnet))
-    except TypeError:
-        hookenv.status_set('waiting', 'Flannel is starting up.')
+@when('flannel.service.started')
+@when_any('cni.is-master', 'flannel.cni.available')
+def ready():
+    ''' Indicate that flannel is active. '''
+    status_set('active', 'Flannel is active.')
+
+
+@when_not('etcd.connected')
+def halt_execution():
+    ''' send a clear message to the user that we are waiting on etcd '''
+    status_set('waiting', 'Waiting for etcd relation.')
 
 
 @hook('upgrade-charm')
 def reset_states_and_redeploy():
     ''' Remove state and redeploy '''
-    remove_state('flannel.host.relayed')
-    remove_state('flannel.installed')
+    remove_state('flannel.binaries.installed')
```

Comment:
Looks like this was handled by removing the `flannel.service.started` state right?